saviSimulation/savi/jason_processing/UxVBehavior.java

```diff
@@ -119,49 +98,79 @@ public void update(double simTime, int perceptionDistance, List<WorldObject> obj
             	double dist = polar.get(Geometry.DISTANCE);
             	if ((azimuth < Math.PI/2. || azimuth > 3* Math.PI/2.)&&(dist <perceptionDistance) ) {
 					//it's visible 
-					visibleItems.add(new CameraPerception(wo.type, this.time, azimuth, elevation, dist));
+					detectedItems.add(new CameraPerception(wo.type, this.time, azimuth, elevation, dist, wo.pixels/2));
+					visibleItems.add(new CameraPerception(wo.type, this.time, azimuth, elevation, dist, wo.pixels/2));
             	}
+			}	   	
+		}
+		
+		//remove objects covered by others on the visualization
+		for(CameraPerception di:detectedItems) {
+			for(int i=0; i<visibleItems.size();i++) {
+				//to calculate visual angle covered by the object
+				distance = di.getParameters().get(2);
+				//angle deviation from centroid = radius
+				oposite = di.getParameters().get(3);
+				//math to calculate angle cover
+				tan=oposite/distance;
+				angle=Math.abs(Math.atan(tan));
+				
+				//if object is covered by di remove
+				//if is covered by azimuth angle
+				if(di.getParameters().get(0)+angle > visibleItems.get(i).getParameters().get(0) && di.getParameters().get(0)-angle < visibleItems.get(i).getParameters().get(0) ) {
+					//if it is covered by elevation angle
+					if(di.getParameters().get(1)+angle > visibleItems.get(i).getParameters().get(1) && di.getParameters().get(1)-angle < visibleItems.get(i).getParameters().get(1) ){
+						//if it's at a mayor distance
+						if(di.getParameters().get(2) < visibleItems.get(i).getParameters().get(2)) {
+							visibleItems.remove(visibleItems.get(i));
+						}
+					}
+				}
 			}	
-            	
 		}
+		
 		return visibleItems;
 	}
 	/**
-	 * Process the action in the queue to update the speedVal and compassAngle
+	 * Wifi Communication
 	 */
-	protected void processAgentActions(){
-		List<String> toexec = agentState.getAllActions();   
-		for (String action : toexec) {
-			System.out.println("[ process actions]UAS id="+this.ID+" doing:"+ action);
-			if (action.equals("turn(left)")) //TODO: make these MOD 2 pi ? 
-				this.compasAngle -= Math.PI/16.0;
-				//Normalize compass angle between 0 and 2 Pi
-				if(compasAngle<0) compasAngle+=2*Math.PI;
-			else if (action.equals("turn(right)"))				
-				this.compasAngle += Math.PI/16.0;
-				//Normalize compass angle between 0 and 2 Pi
-				if(compasAngle>2*Math.PI) compasAngle-=2*Math.PI;
-			else if (action.equals( "thrust(on)")) 
-				this.speedVal = SPEED;
-			else if (action.equals("thrust(off)")) 
-				this.speedVal = 0;  
-		}					
-	}	
 	
-	/**
-	 * Get UAS id
-	 * @return
-	 */
-	public String getID() {		
-		return ID;
+/*	DEPRECATED(?)
+	protected void wifiCommunication(int WIFI_PERCEPTION_DISTANCE, List<WorldObject> objects) {
```

Comment:
yep. can be removed. Unsure why the behaviour needs wifiProbWorking.

Reply:
It is a deprecated thing. I removed it.